diff --git a/Makefile.am b/Makefile.am
index 6eb7b5b..f53a58b 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -269,6 +269,10 @@ if ENABLE_ORACLE
 SUBDIRS += plugins/omoracle
 endif
 
+if ENABLE_MMOPENSHIFT
+SUBDIRS += plugins/mmopenshift
+endif
+
 if ENABLE_GUI
 SUBDIRS += java
 endif
diff --git a/configure.ac b/configure.ac
index c3449c7..bb63e8d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1524,6 +1524,28 @@ AM_CONDITIONAL(ENABLE_OMHIREDIS, test x$enable_omhiredis = xyes)
 
 # END HIREDIS SUPPORT
 
+# OPENSHIFT SUPPORT
+
+AC_ARG_ENABLE(mmopenshift,
+        [AS_HELP_STRING([--enable-mmopenshift],[Enable building mmopenshift support @<:@default=no@:>@])],
+        [case "${enableval}" in
+         yes) enable_mmopenshift="yes" ;;
+          no) enable_mmopenshift="no" ;;
+           *) AC_MSG_ERROR(bad value ${enableval} for --enable-mmopenshift) ;;
+         esac],
+        [enable_mmopenshift=no]
+)
+if test "x$enable_openshift" = "xyes"; then
+  AC_CHECK_HEADERS(
+    [sys/inotify.h],,
+    [AC_MSG_FAILURE([sys/inotify.h is missing])]
+  )
+  AC_CHECK_FUNCS(inotify_init)
+fi
+AM_CONDITIONAL(ENABLE_MMOPENSHIFT, test x$enable_mmopenshift = xyes)
+
+# END OPENSHIFT SUPPORT
+
 AC_CONFIG_FILES([Makefile \
 		runtime/Makefile \
 		compat/Makefile \
@@ -1585,6 +1607,7 @@ AC_CONFIG_FILES([Makefile \
 		plugins/omelasticsearch/Makefile \
 		plugins/sm_cust_bindcdr/Makefile \
 		plugins/mmsnmptrapd/Makefile \
+		plugins/mmopenshift/Makefile \
 		java/Makefile \
 		tests/Makefile])
 AC_OUTPUT
@@ -1649,6 +1672,7 @@ echo "    mmutf8fix enabled:                        $enable_mmutf8fix"
 echo "    mmrfc5424addhmac enabled:                 $enable_mmrfc5424addhmac"
 echo "    mmpstrucdata enabled:                     $enable_mmpstrucdata"
 echo "    mmsequence enabled:                       $enable_mmsequence"
+echo "    mmopenshift module will be compiled:      $enable_mmopenshift"
 echo
 echo "---{ strgen modules }---"
 echo "    sm_cust_bindcdr module will be compiled:  $enable_sm_cust_bindcdr"
